src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs

```diff
@@ -31,19 +31,54 @@ public SelectYourPCSController(Func<IWeeeClient> apiclient, BreadcrumbService br
         }
 
         [HttpGet]
-        public virtual async Task<ActionResult> Index(Guid organisationId)
+        public virtual async Task<ActionResult> Index(Guid organisationId, Guid returnId)
         {
-            var viewModel = new SelectYourPCSViewModel();
+            var viewModel = new SelectYourPCSViewModel()
+            {
+                OrganisationId = organisationId,
+                ReturnId = returnId
+            };
 
             using (var client = apiClient())
             {
-                var schemeRequest = new GetSchemes();
-                schemeList = await client.SendAsync(User.GetAccessToken(), schemeRequest);
-                viewModel.SchemeList = schemeList;
+                viewModel.SchemeList = await client.SendAsync(User.GetAccessToken(), new GetSchemesExternal());
             }
             await SetBreadcrumb(organisationId, BreadCrumbConstant.AatfReturn);
 
-            return View(viewModel);
+            return View("Index", viewModel);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public virtual async Task<ActionResult> Index(SelectYourPCSViewModel viewModel)
+        {
+            if (ModelState.IsValid)
+            {
+                using (var client = apiClient())
+                {
+                    foreach (var scheme in viewModel.SelectedSchemes)
+                    {
+                        var returnSchemeRequest = new AddReturnScheme()
+                        {
+                            SchemeId = scheme,
+                            ReturnId = viewModel.ReturnId
+                        };
+
+                        await client.SendAsync(User.GetAccessToken(), returnSchemeRequest);
+                    }
+                }
+                return RedirectToAction("Index", "AatfTaskList",
+                                            new { area = "AatfReturn", organisationId = viewModel.OrganisationId, returnId = viewModel.ReturnId });
+            }
+            else
+            {
+                using (var client = apiClient())
+                {
+                    viewModel.SchemeList = await client.SendAsync(User.GetAccessToken(), new GetSchemesExternal());
```

Comment:
We shouldnt need to re-retrieve the schemes  list if we have set the view and view model correct as they should get bound by the model binder... 

It should just be a return View(viewModel) when the ModelState isnt valid